src/main/resources/liquibase/crop_changelog/18_3_0.xml

```diff
@@ -104,4 +104,397 @@
+	</changeSet>
+
+	<changeSet author="clarysabel" id="v18.3.0-18">
+		<preConditions onFail="MARK_RAN">
+			<and>
+				<tableExists tableName="udflds_attributes_bkp"></tableExists>
+				<sqlCheck expectedResult="1">
+					SELECT COUNT(*) > 0
+					FROM udflds_attributes_bkp
+					WHERE
+					cvterm_id is not null;
+				</sqlCheck>
+				<sqlCheck expectedResult="1">
+					select count(*) > 0 from listdataprops l inner join udflds_attributes_bkp bkp on bkp.fcode = l.column_name;
+				</sqlCheck>
+			</and>
+		</preConditions>
+		<comment>Replace old attributes names in listdataprops with the new variable names</comment>
+		<sql dbms="mysql">
+			ALTER TABLE listdataprops
+			  ADD COLUMN column_name_tmp varchar(50) DEFAULT NULL;
+
+			UPDATE listdataprops INNER JOIN udflds_attributes_bkp bkp ON bkp.fcode = column_name
+			SET column_name_tmp = bkp.cvterm_name WHERE 1 = 1;
+
+			UPDATE listdataprops SET column_name = UPPER(column_name_tmp) WHERE column_name_tmp IS NOT NULL;
+
+			ALTER TABLE listdataprops DROP COLUMN column_name_tmp;
+		</sql>
+	</changeSet>
+
+	<changeSet author="clarysabel" id="v18.3.0-19">
+		<preConditions onFail="MARK_RAN">
+			<and>
+				<tableExists tableName="udflds_attributes_bkp"></tableExists>
+				<sqlCheck expectedResult="1">
+					SELECT COUNT(*) > 0
+					FROM udflds_attributes_bkp
+					WHERE
+					cvterm_id is not null;
+				</sqlCheck>
+				<sqlCheck expectedResult="0">
+					SELECT
+					COUNT(*)
+					FROM
+					information_schema.routines
+					WHERE
+					routine_schema = DATABASE()
+					AND specific_name = 'replace_attr_process_code_in_methods';
+				</sqlCheck>
+			</and>
+		</preConditions>
+		<comment>Create store procedure replace_attr_process_code_in_methods</comment>
+		<sql dbms="mysql" splitStatements="false">
+			<![CDATA[
+
+				CREATE PROCEDURE replace_attr_process_code_in_methods()
+
+				BEGIN
+
+				  DECLARE done INT DEFAULT FALSE;
+				  DECLARE attrsc_old, attrmp_old, attrfp_old VARCHAR(255);
+				  DECLARE attrsc_new, attrmp_new, attrfp_new VARCHAR(255);
+
+				  DECLARE cur_replace_tmp CURSOR FOR
+					SELECT * FROM codes_to_replace_tmp t WHERE EXISTS (
+						SELECT 1 FROM methods where prefix like CONCAT('%',t.ATTRSC_OLD,'%') or prefix like CONCAT('%',t.ATTRFP_OLD,'%')
+						or prefix like CONCAT('%',t.ATTRMP_OLD,'%') or suffix like CONCAT('%',t.ATTRSC_OLD,'%') or suffix like CONCAT('%',t.ATTRFP_OLD,'%')
+						or suffix like CONCAT('%',t.ATTRMP_OLD,'%')
+					);
+
+				  DECLARE CONTINUE HANDLER FOR NOT FOUND SET done = TRUE;
+
+				  CREATE TEMPORARY TABLE codes_to_replace_tmp
+					SELECT CONCAT('[ATTRSC.', fcode, ']') as ATTRSC_OLD,
+						   CONCAT('[ATTRFP.', fcode, ']') as ATTRFP_OLD,
+						   CONCAT('[ATTRMP.', fcode, ']') as ATTRMP_OLD,
+						   CONCAT('[ATTRSC.', cvterm_id, ']') as ATTRSC_NEW,
+						   CONCAT('[ATTRFP.', cvterm_id, ']') as ATTRFP_NEW,
+						   CONCAT('[ATTRMP.', cvterm_id, ']') as ATTRMP_NEW
+					FROM udflds_attributes_bkp where ftype = 'ATTRIBUTE';
```

Comment:
I don't recall much now about attribute process codes but are we sure that only records ftype = 'ATTRIBUTE' is possible to be selected for methods configuration for prefix and suffix?

Reply:
Yes, it is only defined for prefix and suffix as it is documented in 
https://github.com/IntegratedBreedingPlatform/Documentation/wiki/Process-Codes-Available-for-Use-in-Name-Generation